zds/middlewares/matomomiddleware.py

```diff
@@ -0,0 +1,95 @@
+from _datetime import datetime
+from queue import Queue
+
+import requests
+import logging
+
+from django.conf import settings
+from threading import Thread
+
+matomo_api_url = "{}/matomo.php".format(settings.ZDS_APP["site"]["matomoUrl"])
+matomo_site_id = settings.ZDS_APP["site"]["matomoSiteID"]
+matomo_api_version = 1
+logger = logging.getLogger(__name__)
+tracked_status_code = [200]
+tracked_methods = ["GET"]
+excluded_paths = ["/contenus", "/mp", "/munin", "/api"]
+
+
+def _background_process(queue: Queue):
+    data = queue.get(block=True)
+    while data:
+        params = {
+            "idsite": matomo_site_id,
+            "action_name": data["r_path"],
+            "rec": 1,
+            "apiv": matomo_api_version,
+            "lang": data["client_accept_language"],
+            "ua": data["client_user_agent"],
+            "urlref": data["client_referer"],
+            "url": data["client_url"],
+            "h": data["datetime"].hour,
+            "m": data["datetime"].minute,
+            "s": data["datetime"].second,
+        }
+        try:
+            requests.get(
+                matomo_api_url,
+                params=params,
+            )
+            logger.info(f'Matomo tracked this link : {data["client_url"]}')
+        except Exception as e:
+            logger.error(f'Something went wrong with the tracking of the link {data["client_url"]}')
+            logger.error(f"{str(e)}")
```

Comment:
Ce petit changement permet d'avoir ceci dans Sentry :

![image](https://user-images.githubusercontent.com/6664636/114317386-1e942100-9b08-11eb-8e9b-43c1d8356b48.png)

Au lieu de ceci : 

![image](https://user-images.githubusercontent.com/6664636/114317398-2e136a00-9b08-11eb-9662-5c41d377cd5d.png)

```suggestion
        except Exception:
            logger.error(f'Something went wrong with the tracking of the link {data["client_url"]}')
```

Reply:
À noter qu'on garde la trace de l'erreur sous-jacente quand on ouvre le ticket : 

![image](https://user-images.githubusercontent.com/6664636/114317509-b3971a00-9b08-11eb-908f-73617d7b321e.png)